transaction/tests/AccessMode_unittest.cpp

```diff
@@ -0,0 +1,114 @@
+
+  EXPECT_TRUE(is_mode.isCompatible(is_mode));
+  EXPECT_TRUE(is_mode.isCompatible(ix_mode));
+  EXPECT_TRUE(is_mode.isCompatible(s_mode));
+  EXPECT_FALSE(is_mode.isCompatible(x_mode));
+
+  EXPECT_TRUE(ix_mode.isCompatible(is_mode));
+  EXPECT_TRUE(ix_mode.isCompatible(ix_mode));
+  EXPECT_FALSE(ix_mode.isCompatible(s_mode));
+  EXPECT_FALSE(ix_mode.isCompatible(x_mode));
+
+  EXPECT_TRUE(s_mode.isCompatible(is_mode));
+  EXPECT_FALSE(s_mode.isCompatible(ix_mode));
+  EXPECT_TRUE(s_mode.isCompatible(s_mode));
+  EXPECT_FALSE(s_mode.isCompatible(x_mode));
+
+  EXPECT_FALSE(x_mode.isCompatible(is_mode));
+  EXPECT_FALSE(x_mode.isCompatible(ix_mode));
+  EXPECT_FALSE(x_mode.isCompatible(s_mode));
+  EXPECT_FALSE(x_mode.isCompatible(x_mode));
+}
+
+TEST(AccessModeTest, ModeQueryChecks) {
+  AccessMode is_mode = AccessMode(AccessModeType::kIsLock);
+  AccessMode ix_mode = AccessMode(AccessModeType::kIxLock);
+  AccessMode s_mode = AccessMode(AccessModeType::kSLock);
+  AccessMode x_mode = AccessMode(AccessModeType::kXLock);
+
+  EXPECT_TRUE(is_mode.isIntentionShareLock());
+  EXPECT_FALSE(is_mode.isExclusiveLock());
+  EXPECT_FALSE(is_mode.isShareLock());
+  EXPECT_FALSE(is_mode.isExclusiveLock());
+
+  EXPECT_FALSE(ix_mode.isIntentionShareLock());
+  EXPECT_TRUE(ix_mode.isIntentionExclusiveLock());
+  EXPECT_FALSE(ix_mode.isShareLock());
+  EXPECT_FALSE(ix_mode.isExclusiveLock());
+
+  EXPECT_FALSE(s_mode.isIntentionShareLock());
+  EXPECT_FALSE(s_mode.isIntentionExclusiveLock());
+  EXPECT_TRUE(s_mode.isShareLock());
+  EXPECT_FALSE(s_mode.isExclusiveLock());
+
+  EXPECT_FALSE(x_mode.isIntentionShareLock());
+  EXPECT_FALSE(x_mode.isIntentionExclusiveLock());
+  EXPECT_FALSE(x_mode.isShareLock());
+  EXPECT_TRUE(x_mode.isExclusiveLock());
+}
+
+TEST(AccessModeTest, Equality) {
+  AccessMode is_mode_1 = AccessMode(AccessModeType::kIsLock);
+  AccessMode ix_mode_1 = AccessMode(AccessModeType::kIxLock);
+  AccessMode s_mode_1 = AccessMode(AccessModeType::kSLock);
+  AccessMode x_mode_1 = AccessMode(AccessModeType::kXLock);
+
+  AccessMode is_mode_2 = AccessMode(AccessModeType::kIsLock);
+  AccessMode ix_mode_2 = AccessMode(AccessModeType::kIxLock);
+  AccessMode s_mode_2 = AccessMode(AccessModeType::kSLock);
+  AccessMode x_mode_2 = AccessMode(AccessModeType::kXLock);
+
+  EXPECT_TRUE(is_mode_1 == is_mode_2);
+  EXPECT_FALSE(is_mode_1 == ix_mode_1);
+  EXPECT_FALSE(is_mode_1 == s_mode_1);
+  EXPECT_FALSE(is_mode_1 == x_mode_1);
+
+  EXPECT_FALSE(ix_mode_1 == is_mode_1);
+  EXPECT_TRUE(ix_mode_1 == ix_mode_2);
+  EXPECT_FALSE(ix_mode_1 == s_mode_1);
+  EXPECT_FALSE(ix_mode_1 == x_mode_1);
+
+  EXPECT_FALSE(s_mode_1 == is_mode_1);
+  EXPECT_FALSE(s_mode_1 == ix_mode_1);
+  EXPECT_TRUE(s_mode_1 == s_mode_2);
+  EXPECT_FALSE(s_mode_1 == x_mode_1);
+
+  EXPECT_FALSE(x_mode_1 == is_mode_1);
+  EXPECT_FALSE(x_mode_1 == ix_mode_1);
+  EXPECT_FALSE(x_mode_1 == s_mode_1);
+  EXPECT_TRUE(x_mode_1 == x_mode_2);
```

Comment:
Try to replace `EXPECT_TRUE` with `EXPECT_EQ` (the left-hand side is the expected value, and the right-hand is the checked), `EXPECT_FALSE` with `EXPECT_NE`.